DataLayer/MpptData/MpptData.cpp

```diff
@@ -3,42 +3,20 @@
 
 MpptData::MpptData()
 {
+    mpptList_.reserve(3);
 }
 
 MpptData::~MpptData()
 {
 }
 
-Mppt MpptData::mpptZero() const
+Mppt MpptData::getMppt(int i) const
 {
-    return mpptList[0];
+    return mpptList_.at(i);
 }
 
-Mppt MpptData::mpptOne() const
+void MpptData::setMppt(int i, Mppt mppt)
 {
-    return mpptList[1];
-}
-
-Mppt MpptData::mpptTwo() const
-{
-    return mpptList[2];
-}
-
-
-void MpptData::setMpptZero(Mppt mpptZero)
-{
-    mpptList[0] = mpptZero;
-    emit mpptZeroReceived(mpptList[0]);
-}
-
-void MpptData::setMpptOne(Mppt mpptOne)
-{
-    mpptList[1] = mpptOne;
-    emit mpptOneReceived(mpptList[1]);
-}
-
-void MpptData::setMpptTwo(Mppt mpptTwo)
-{
-    mpptList[2] = mpptTwo;
-    emit mpptTwoReceived(mpptList[2]);
+    mpptList_.replace(i, mppt);
+    emit mpptReceived(i, mpptList_.at(i));
```

Comment:
Note that in `MpptPresenter.cpp` there are a bunch of connects that go with the ones you replaced.